crypto/ec/ecp_nistp384.c

```diff
@@ -0,0 +1,1935 @@
+ {1, 0, 0, 0, 0, 0, 0}},
+{{0x00430155e3b908af, 0x00b801e4fec25226, 0x00b0d4bcfe806d26, 0x009fc4014eb13d37,
+  0x0066c94e44ec07e8, 0x00d16adc03874ba2, 0x000030c917a0d2a7},
+ {0x00edac9e21eb891c, 0x00ef0fb768102eff, 0x00c088cef272a5f3, 0x00cbf782134e2964,
+  0x0001044a7ba9a0e3, 0x00e363f5b194cf3c, 0x00009ce85249e372},
+ {1, 0, 0, 0, 0, 0, 0}},
+{{0x001dd492dda5a7eb, 0x008fd577be539fd1, 0x002ff4b25a5fc3f1, 0x0074a8a1b64df72f,
+  0x002ba3d8c204a76c, 0x009d5cff95c8235a, 0x0000e014b9406e0f},
+ {0x008c2e4dbfc98aba, 0x00f30bb89f1a1436, 0x00b46f7aea3e259c, 0x009224454ac02f54,
+  0x00906401f5645fa2, 0x003a1d1940eabc77, 0x00007c9351d680e6},
+ {1, 0, 0, 0, 0, 0, 0}},
+{{0x005a35d872ef967c, 0x0049f1b7884e1987, 0x0059d46d7e31f552, 0x00ceb4869d2d0fb6,
+  0x00e8e89eee56802a, 0x0049d806a774aaf2, 0x0000147e2af0ae24},
+ {0x005fd1bd852c6e5e, 0x00b674b7b3de6885, 0x003b9ea5eb9b6c08, 0x005c9f03babf3ef7,
+  0x00605337fecab3c7, 0x009a3f85b11bbcc8, 0x0000455470f330ec},
+ {1, 0, 0, 0, 0, 0, 0}},
+{{0x002197ff4d55498d, 0x00383e8916c2d8af, 0x00eb203f34d1c6d2, 0x0080367cbd11b542,
+  0x00769b3be864e4f5, 0x0081a8458521c7bb, 0x0000c531b34d3539},
+ {0x00e2a3d775fa2e13, 0x00534fc379573844, 0x00ff237d2a8db54a, 0x00d301b2335a8882,
+  0x000f75ea96103a80, 0x0018fecb3cdd96fa, 0x0000304bf61e94eb},
+ {1, 0, 0, 0, 0, 0, 0}},
+{{0x00b2afc332a73dbd, 0x0029a0d5bb007bc5, 0x002d628eb210f577, 0x009f59a36dd05f50,
+  0x006d339de4eca613, 0x00c75a71addc86bc, 0x000060384c5ea93c},
+ {0x00aa9641c32a30b4, 0x00cc73ae8cce565d, 0x00ec911a4df07f61, 0x00aa4b762ea4b264,
+  0x0096d395bb393629, 0x004efacfb7632fe0, 0x00006f252f46fa3f},
+ {1, 0, 0, 0, 0, 0, 0}},
+{{0x00567eec597c7af6, 0x0059ba6795204413, 0x00816d4e6f01196f, 0x004ae6b3eb57951d,
+  0x00420f5abdda2108, 0x003401d1f57ca9d9, 0x0000cf5837b0b67a},
+ {0x00eaa64b8aeeabf9, 0x00246ddf16bcb4de, 0x000e7e3c3aecd751, 0x0008449f04fed72e,
+  0x00307b67ccf09183, 0x0017108c3556b7b1, 0x0000229b2483b3bf},
+ {1, 0, 0, 0, 0, 0, 0}},
+{{0x00e7c491a7bb78a1, 0x00eafddd1d3049ab, 0x00352c05e2bc7c98, 0x003d6880c165fa5c,
+  0x00b6ac61cc11c97d, 0x00beeb54fcf90ce5, 0x0000dc1f0b455edc},
+ {0x002db2e7aee34d60, 0x0073b5f415a2d8c0, 0x00dd84e4193e9a0c, 0x00d02d873467c572,
+  0x0018baaeda60aee5, 0x0013fb11d697c61e, 0x000083aafcc3a973},
+ {1, 0, 0, 0, 0, 0, 0}}
+};
+
+/*
+ * select_point selects the |idx|th point from a precomputation table and
+ * copies it to out.
+ */
+ /* pre_comp below is of the size provided in |size| */
+static void select_point(const limb idx, unsigned int size,
+                         const felem pre_comp[][3], felem out[3])
+{
+    unsigned i, j;
+    limb *outlimbs = &out[0][0];
+
+    memset(out, 0, sizeof(*out) * 3);
+
+    for (i = 0; i < size; i++) {
+        const limb *inlimbs = &pre_comp[i][0][0];
+        limb mask = i ^ idx;
+        mask |= mask >> 4;
+        mask |= mask >> 2;
+        mask |= mask >> 1;
+        mask &= 1;
+        mask--;
+        for (j = 0; j < NLIMBS * 3; j++)
+            outlimbs[j] |= inlimbs[j] & mask;
```

Comment:
Again, would our constant time support be useful here?

Reply:
As `select_point` is used in a hot path, and so halves the performance of generating digital signatures unfortunately. Keeping original implementation.